sdk/formrecognizer/ai-form-recognizer/src/lro/analyze.ts

```diff
@@ -0,0 +1,242 @@
+   */
+  modelId: string;
+
+  /**
+   * The unique ID of this analysis operation.
+   */
+  operationId: string;
+}
+
+/**
+ * A long-running operation (poller) that tracks the state of an analysis operation, eventually producing the result
+ * type that corresponds to the model.
+ */
+export type AnalysisPoller<Result = AnalyzeResult<AnalyzedDocument>> = PollerLike<
+  DocumentAnalysisPollOperationState<Result>,
+  Result
+>;
+
+const operationLocationRegex = /\/documentModels\/([a-zA-Z0-9-]+)\/analyzeResults\/([a-z0-9-]+)/;
+
+/**
+ * Extract a model ID and analysis operation ID from an operationLocation URL.
+ * @internal
+ */
+export function parseOperationLocation(url: string | undefined): [string, string] {
+  if (url === undefined) {
+    throw new Error("Failed to start analysis operation: no operation-location in the response.");
+  }
+
+  const parseResult = operationLocationRegex.exec(url);
+  if (!parseResult || !parseResult[1] || !parseResult[2]) {
+    throw new Error(`Unable to parse operationLocation: "${url}"`);
+  }
+
+  return [parseResult[1], parseResult[2]];
+}
+
+/**
+ * Convert a generated AnalyzeResult into a convenience layer AnalyzeResult.
+ * @internal
+ */
+export function toAnalyzeResultFromGenerated<
+  Mapper extends (document: GeneratedDocument) => unknown
+>(
+  result: GeneratedAnalyzeResult,
+  mapDocuments: Mapper
+): AnalyzeResult<Mapper extends (...args: never) => infer Document ? Document : never> {
+  type Document = Mapper extends (...args: never) => infer ThisDocument ? ThisDocument : never;
+  return {
+    ...result,
+    apiVersion: result.apiVersion as FormRecognizerApiVersion,
+    tables: result.tables ?? [],
+    keyValuePairs: result.keyValuePairs ?? [],
+    entities: result.entities ?? [],
+    styles: result.styles ?? [],
+    documents: (result.documents?.map((doc) => mapDocuments(doc)) as Document[]) ?? [],
+  };
+}
+
+/**
+ * Defines an analysis operation by the initial request model ID, poller options, and a transform to apply to the
+ * eventual result,
+ *
+ * @internal
+ */
+export interface AnalysisOperationDefinition<Result = AnalyzeResult> {
+  transformResult: (primitiveResult: GeneratedAnalyzeResult) => Result;
+  initialModelId: string;
+  options: PollerOptions<DocumentAnalysisPollOperationState<Result>> &
+    AnalyzeDocumentsOptions<Result>;
+}
+
+/**
+ * Converts an AnalyzeResultOperation (LRO response) to a DocumentAnalysisPollOperationState
+ * @internal
+ */
+export function toDocumentAnalysisPollOperationState<Result>(
+  definition: AnalysisOperationDefinition<Result>,
+  modelId: string,
+  operationId: string,
+  result: AnalyzeResultOperation
+): DocumentAnalysisPollOperationState<Result> {
+  return {
+    status: result.status,
+    modelId: modelId,
+    operationId,
+    result: result.analyzeResult && definition.transformResult(result.analyzeResult),
+    error: result.error && new FormRecognizerError(result.error),
+    isCancelled: false, // Not supported
```

Comment:
What happens if the `cancel` method get called?

Reply:
The poller implementation will throw an exception.